app/src/main/java/cm/aptoide/pt/home/BundlesRepository.java

```diff
@@ -0,0 +1,80 @@
+package cm.aptoide.pt.home;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.List;
+import rx.Single;
+
+/**
+ * Created by jdandrade on 08/03/2018.
+ */
+
+public class BundlesRepository {
+  private final BundleDataSource remoteBundleDataSource;
+  private AbstractMap.SimpleEntry<Integer, List<HomeBundle>> cachedBundles;
+
+  public BundlesRepository(BundleDataSource remoteBundleDataSource,
+      AbstractMap.SimpleEntry<Integer, List<HomeBundle>> cachedBundles) {
+    this.remoteBundleDataSource = remoteBundleDataSource;
+    this.cachedBundles = cachedBundles;
+  }
+
+  public Single<HomeBundlesModel> loadHomeBundles() {
+    int limit = 5;
```

Comment:
agree, this needs to be refactored, I will extract it to a field to be injected from the provider, In the future in case we need to change for something more dynamic we change.